source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,931 @@
+    buffCfg[ 0 ].txBufferAlign = &( txDataBuff_0[ 0 ][ 0 ] ); /* Transmit data buffer start address. */
+    buffCfg[ 0 ].txFrameInfo = NULL;                          /* Transmit frame information start address. Set only if using zero-copy transmit. */
+    buffCfg[ 0 ].rxMaintainEnable = true;                     /* Receive buffer cache maintain. */
+    buffCfg[ 0 ].txMaintainEnable = true;                     /* Transmit buffer cache maintain. */
+
+    sysClock = phyHandle.mdioHandle->resource.csrClock_Hz;
+
+    ENET_GetDefaultConfig( &config );
+
+    config.ringNum = ENET_RING_NUM;
+    config.rxBuffAlloc = NULL;
+    config.rxBuffFree = NULL;
+    config.userData = ethernetifLocal;
+    config.miiSpeed = ( enet_mii_speed_t ) speed;
+    config.miiDuplex = ( enet_mii_duplex_t ) duplex;
+
+    /* Only get interrupt for incoming messages. */
+    config.interrupt = kENET_RxFrameInterrupt;
+    config.callback = ethernet_callback;
+
+    for( instance = 0; instance < ARRAY_SIZE( enetBases ); instance++ )
+    {
+        if( enetBases[ instance ] == ethernetifLocal->base )
+        {
+            NVIC_SetPriority( enetRxIrqId[ instance ], ENET_PRIORITY );
+            NVIC_SetPriority( enetTxIrqId[ instance ], ENET_PRIORITY );
+            break;
+        }
+    }
+
+    configASSERT( instance != ARRAY_SIZE( enetBases ) );
+
+    if( instance == ARRAY_SIZE( enetBases ) )
+    {
+        xStatus = kStatus_Fail;
+    }
+    else
+    {
+        pxEndPoint = FreeRTOS_FirstEndPoint( pxMyInterface );
+        configASSERT( pxEndPoint != NULL );
+
+        for( i = 0; i < ENET_RXBUFF_NUM; i++ )
+        {
+            ethernetifLocal->RxPbufs[ i ].buffer = &( ethernetifLocal->RxDataBuff[ i ][ 0 ] );
+            ethernetifLocal->RxPbufs[ i ].buffer_used = false;
+        }
+
+        /* Initialize the ENET module. */
+        xStatus = ENET_Init( ethernetifLocal->base,
+                             &ethernetifLocal->handle,
+                             &config,
+                             &buffCfg[ 0 ],
+                             pxEndPoint->xMACAddress.ucBytes,
+                             sysClock );
+
+        #if ( ipconfigUSE_LLMNR == 1 )
+            ENET_AddMulticastGroup( ethernetifLocal->base, ( uint8_t * ) xLLMNR_MacAdress.ucBytes );
+        #endif /* ipconfigUSE_LLMNR */
+
+        #if ( ipconfigUSE_IPv6 != 0 )
+            #if ( ipconfigUSE_LLMNR == 1 )
+                ENET_AddMulticastGroup( ethernetifLocal->base, ( uint8_t * ) xLLMNR_MacAdressIPv6.ucBytes );
+            #endif /* ipconfigUSE_LLMNR */
+
+            for( pxEndPoint = FreeRTOS_FirstEndPoint( pxMyInterface );
+                 pxEndPoint != NULL;
+                 pxEndPoint = FreeRTOS_NextEndPoint( pxMyInterface, pxEndPoint ) )
+            {
+                if( pxEndPoint->bits.bIPv6 != pdFALSE_UNSIGNED )
+                {
+                    uint8_t ucMACAddress[ 6 ] = { 0x33, 0x33, 0xff, 0, 0, 0 };
```

Comment:
It would be great if we could add some comments to explain that we are adding solicited-node multicast MAC addresses which is used for Duplicate Address Detection to the allow list of MAC addresses, as its not very evident from the variable names what kind of MAC addresses are these, similar to the case of LLMNR MACs.